### ISSUE TITLE
Also sort the exports

### ISSUE THREAD

**Issue by eduardodallmann** [IanVS/prettier-plugin-sort-imports#119]
**Is your feature request related to a problem?**
Sort also export not just imports.

**Describe the solution you'd like**
When I have code like that:

```
export { c, a, b, A } from 'foo';
export { x } from 'bar';
```
To be sorted like that:

```
export { x } from 'bar';
export { A, a, b, c } from 'foo';
```

**Describe alternatives you've considered**
None

**Additional context**
Related: https://github.com/trivago/prettier-plugin-sort-imports/issues/159

**Comment by fbartho** [comment]
@eduardodallmann unfortunately sorting exports could break code. For example:

```typescript
export const FooAPI= new MyAPI(args);

// export our default implementation separately
export const beAwesome = FooAPI.beAwesome:
``` 

If you sorted this, your output would break! Obviously this is a trivial example, there’s tons of other ways to case moving around exports to be a problem.

this is a neat idea, that might make sense in someone’s codebase, so I encourage you to to make a prettier-plug-in-sort-exports library. I just think that’s out-of-scope for this library!

**Comment by eduardodallmann** [comment]
You are correct. Sorting export lines can cause a lot of problems. But I believe that organizing just the de-structuring could be good.

This plugin does that https://github.com/simonhaenisch/prettier-plugin-organize-imports

**Comment by eduardodallmann** [comment]
I'm currently using the prettier-plugin-sort-imports lib. In my opinion it is much better because it is more like trivago but solves many of the problems.

The only thing I like about prettier-plugin-organize-imports is organizing the export.

I'm in a remix.run project, and by convention the routes have loader and action functions being exported. But to better separate and organize the codebase I did it this way:

```
routes/
├── login/
│   ├── login.server.ts
│   ├── route.tsx
```
login.server.ts
```
export const loader: LoaderFunction = (): LoginLoaderReturn => ....

export const action: ActionFunction = async ({ request }) => ....
```

route.tsx
```
import ....
import ....

export { loader, action } from './login.server';

export default function Login() {
   return a react component....
```
Reference https://remix.run/docs/en/1.19.3/route/loader

**Comment by SalahAdDin** [comment]
I'm looking exactly for something like this, but there are not maintained libraries at the moment, and honestly, I don't understand why.

**Comment by IanVS** [comment]
@SalahAdDin, are you looking for sorting exports, or only re-exports from other files?

**Comment by SalahAdDin** [comment]
> @SalahAdDin, are you looking for sorting exports, or only re-exports from other files?

Sorting exports, We don't have to go every file and sort them manually hahahahaha.

**Comment by IanVS** [comment]
I mean, do you want to sort all exports?  If a file exports two functions and a constant, you want to sort those?  Or only sort re-exports like this?

```ts
export foo from "./foo";
```

**Comment by SalahAdDin** [comment]
> I mean, do you want to sort all exports? If a file exports two functions and a constant, do you want to sort those? Or only sort re-exports like this?
> 
> ```ts
> export foo from "./foo";
> ```

Not re-export, the file exports, that is the most common use for the export function (i think).

**Comment by IanVS** [comment]
Can you give an example of a before and after?

**Comment by SalahAdDin** [comment]
> Can you give an example of a before and after?

**Input:**
```
export type {
  RContentNode,
  RImage,
  RContent,
  RLink,
  RList,
  RNode,
  RTextAttributes,
  RListItem,
  RVideo,,
  RNodeChildren,
  RText,
  RNodeType
};
```

**Ouput:**
```
export type {
  RContent,
  RContentNode,
  RImage,
  RLink,
  RList,
  RListItem,
  RNode,
  RNodeChildren,
  RNodeType,
  RText,
  RTextAttributes,
  RVideo,
};
```

**Comment by IanVS** [comment]
ok, so just the specifiers within a named export block, got it.  I think that could potentially be doable, but it would need to be placed behind a new option, defaulted to false, to avoid unexpected changes.

I don't have time to work on it right now, but I'd be willing to review a PR if someone else wants to take a shot.  If anyone plans to work on it, let us know and we can discuss the details.

**Comment by curiosbasant** [comment]
Yes, re-exports should also be sorted just like `imports`

**Comment by babakfp** [comment]
I tried using `@ianvs/prettier-plugin-sort-imports` with `prettier-plugin-organize-imports` but things didn't work as expected.

With this order, `prettier-plugin-organize-imports` nothing happens:

```js
plugins: [
  "prettier-plugin-organize-imports",
  "@ianvs/prettier-plugin-sort-imports",
],
```

When reversing the order, `prettier-plugin-organize-imports` formats over what `@ianvs/prettier-plugin-sort-imports` is formatted.

It could be a good solution if  `@ianvs/prettier-plugin-sort-imports` used `prettier-plugin-organize-imports` for things that it doesn't support, like exports.

**Comment by IanVS** [comment]
>Yes, re-exports should also be sorted just like imports

@curiosbasant Can you give a before/after example of what you want to see?

>It could be a good solution if @ianvs/prettier-plugin-sort-imports used prettier-plugin-organize-imports for things that it doesn't support, like exports.

Prettier doesn't have a great API for multiple plugins to operate on the AST together.  Until they release one, I don't think there's much we can do in this regard.

**Comment by curiosbasant** [comment]
> Can you give a before/after example of what you want to see?

Yes,

_before_
```js
export * from './b'
export { b, a, d } from './c'
export { a, c, b } from './a'
```

_after_
```js
export { a, b, c } from './a'
export * from './b'
export { a, b, d } from './c'
```

This is something that already works in VSCode with the `editor.action.organizeImports` command.

**Comment by SalahAdDin** [comment]
> This is something that already works in VSCode with the `editor.action.organizeImports` command.

Where can we find it? I am unable to see it in settings.

**Comment by curiosbasant** [comment]
> Where can we find it? I am unable to see it in settings.

@SalahAdDin It's not a setting but rather a command that you can trigger from the command palette named `Organize Imports`.

**Comment by cowboy-bebug** [comment]
Hey @SalahAdDin, add this in your `settings.json`:
```json
  "editor.codeActionsOnSave": {
    "source.organizeImports": "always"
  },
```
